src/components/popover/popover.ts

```diff
@@ -61,15 +61,51 @@ export class MdlPopoverComponent implements AfterViewInit {
     }
 
     private hideAllPopovers() {
-        [].map.call(
-          document.querySelectorAll('.mdl-popover.is-visible'),
-          (el: Element) => el.dispatchEvent(new Event('hide'))
-        );
+      let nodeList = document.querySelectorAll('.mdl-popover.is-visible');
+      for(let i=0; i < nodeList.length;++i) {
+        nodeList[i].dispatchEvent(new Event('hide'));
+      }
     }
 
     private show(event: Event) {
         event.stopPropagation();
         this.isVisible = true;
+        this.updateDirection(event);
+    }
+
+    private updateDirection(event: Event) {
+      // To prevent toggle test from failing, since jasmines SpyObj EventTarget didn't work as expected
+      if(!event.target) {
+        return;
+      }
+      const nativeEl = this.elementRef.nativeElement;
+      const targetRect = (<HTMLElement>event.target).getBoundingClientRect();
+      const viewHeight = window.innerHeight;
+
+      // In case something goes wrong, pollBuffer is there to stop avoid memory leak
+      let pollBuffer = 0;
+      let longPoll = setInterval(() => {
```

Comment:
>I checked that single setTimeout(() => { ... } does the job too, why did you add loop?

In my understanding when using setTimeout you are relaying on a 400ms window of time for the operation to be completed ( for angular to digest the change and re-render the component ). So potentially there are 2 scenarios depending on the machine: 
1. when the changes are digested before 400ms timeout, so you end up with visible direction switching
2. when the changes are digested after 400ms timeout and you end up with 0 height on event.target

That was the reasoning behind using interval and clearing interval after the job is done.

Regarding node-dimesions package, I think it will be an overkill to use the another package just for this.